diff --git a/node_modules/recyclerlistview/dist/reactnative/core/StickyContainer.js b/node_modules/recyclerlistview/dist/reactnative/core/StickyContainer.js
index 5ae594d..b052a30 100644
--- a/node_modules/recyclerlistview/dist/reactnative/core/StickyContainer.js
+++ b/node_modules/recyclerlistview/dist/reactnative/core/StickyContainer.js
@@ -65,8 +65,8 @@ var StickyContainer = /** @class */ (function (_super) {
         };
         _this._getRecyclerRef = function (recycler) {
             _this._recyclerRef = recycler;
-            if (_this.props.children.ref) {
-                if (typeof _this.props.children.ref === "function") {
+            if (_this.props.children.props.ref || _this.props.children.ref) {
+                if (typeof _this.props.children.props.ref || _this.props.children.ref === "function") {
                     (_this.props.children).ref(recycler);
                 }
                 else {
diff --git a/node_modules/recyclerlistview/dist/web/core/StickyContainer.js b/node_modules/recyclerlistview/dist/web/core/StickyContainer.js
index 5ae594d..b052a30 100644
--- a/node_modules/recyclerlistview/dist/web/core/StickyContainer.js
+++ b/node_modules/recyclerlistview/dist/web/core/StickyContainer.js
@@ -65,8 +65,8 @@ var StickyContainer = /** @class */ (function (_super) {
         };
         _this._getRecyclerRef = function (recycler) {
             _this._recyclerRef = recycler;
-            if (_this.props.children.ref) {
-                if (typeof _this.props.children.ref === "function") {
+            if (_this.props.children.props.ref || _this.props.children.ref) {
+                if (typeof _this.props.children.props.ref || _this.props.children.ref === "function") {
                     (_this.props.children).ref(recycler);
                 }
                 else {
